d3a_api_client/redis_market.py

```diff
@@ -37,16 +83,79 @@ def _command_received(msg):
         return _command_received
 
     def _subscribe_to_response_channels(self):
-        channel_subs = {f"{self.area_id}/response/{command_name}":
+        channel_subs = {f"{self._channel_prefix}/response/{command_name}":
                         self._generate_command_response_callback(command_name)
                         for command_name in ["market_stats", "grid_fees", "dso_market_stats"]}
         channel_subs.update({
-            f'{self.area_id}/market-events/market': self._on_market_cycle,
-            f'{self.area_id}/events/finish': self._on_finish
+            f'{self._channel_prefix}/response/register_participant': self._on_register,
+            f'{self._channel_prefix}/response/unregister_participant': self._on_unregister,
+            f'{self._channel_prefix}/market-events/market': self._on_market_cycle,
+            f'{self._channel_prefix}/events/finish': self._on_finish
         })
+
         self.pubsub.subscribe(**channel_subs)
         self.pubsub.run_in_thread(daemon=True)
 
+    def _check_transaction_id_cached_out(self, transaction_id):
+        return transaction_id in self._transaction_id_buffer
+
+    def _on_register(self, msg):
+        message = json.loads(msg["data"])
+        self._check_buffer_message_matching_command_and_id(message)
+        self.area_uuid = message["device_uuid"]
+        self.is_active = True
+
+    def _on_unregister(self, msg):
+        message = json.loads(msg["data"])
+        self._check_buffer_message_matching_command_and_id(message)
+        self.is_active = False
+
+    def _check_buffer_message_matching_command_and_id(self, message):
+        if "transaction_id" in message and message["transaction_id"] is not None:
+            transaction_id = message["transaction_id"]
+            if not any(command in ["register", "unregister"] and "transaction_id" in data and
+                       data["transaction_id"] == transaction_id
+                       for command, data in self._blocking_command_responses.items()):
+                raise RedisAPIException("There is no matching command response in _blocking_command_responses.")
+        else:
+            raise RedisAPIException(
+                "the answer message does not contain a valid 'transaction_id' member.")
```

Comment:
```suggestion
                "The targeted message does not contain a valid 'transaction_id' member.")
```

Reply:
Hmm, can you please explain why you think this should be changed? IMO it is indeed a answer message (btw, i implemented this half a year ago, this is copy paste from the device client)

Reply:
Okay, ignore it but you could start Exception messages with Capital letter, and sorry if you feel like me being linguist, my english isn't good either :-)